<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>
  <arg name="localization_rviz" default="false"/>
  <arg name="pnc_rviz" default="true"/>

  <!-- amcl -->
  <include file="$(find xju_slam)/launch/localization.launch">
    <arg name="rviz" value="$(arg localization_rviz)" />
  </include>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find xju_pnc)/param/costmap_params.yaml" command="load" />
    <rosparam file="$(find xju_pnc)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find xju_pnc)/param/teb_local_planner_params.yaml" command="load" />
    <rosparam file="$(find xju_pnc)/param/costmap_converter_params.yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>

  <!-- rviz -->
  <group if="$(arg pnc_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find xju_pnc)/rviz/teb.rviz" />
  </group>
</launch>
